frontend/catalyst/pennylane_extensions.py

```diff
@@ -2250,3 +2252,211 @@ def _callee():
+        )
+
+    if not all(isinstance(l, int) for l in tree_leaves(out_axes)):
+        raise TypeError(
+            f"Invalid 'out_axes'' it can be a PyTree with integer leaves, but got {out_axes}"
+        )
+
+    def batched_fn(*args, **kwargs):
+        """Vectorization wrapper around the hybrid program using catalyst.for_loop"""
+
+        # Check the validity of the input arguments
+        if isinstance(in_axes, tuple) and len(in_axes) != len(args):
+            raise TypeError(
+                "Invalid 'in_axes'; it can be an int or a tuple of PyTrees "
+                f"corresponding to the arguments, but got {len(in_axes)} and {len(args)}"
+            )
+
+        args_flat, args_tree = tree_flatten(args)
+        in_axes_flat = _vmap_tree_flatten(in_axes)
+
+        # import pdb
+        # pdb.set_trace()
+
+        batch_size = _get_batch_size(args_flat, in_axes_flat, axis_size)
+        batch_loc = _get_batch_loc(in_axes_flat)
+
+        # If vmap is requested on any axes
+        if len(batch_loc):
+            for loc in batch_loc:
+                # Support vectorization on dim > 0
+                if ax := in_axes_flat[loc]:
+                    up_axes = [*range(args_flat[loc].ndim)]
+                    up_axes[ax], up_axes[0] = up_axes[0], up_axes[ax]
+                    args_flat[loc] = jnp.transpose(args_flat[loc], up_axes)
+
+        # Prepare args_flat to run 'fn' one time and get the output-shape
+        fn_args_flat = args_flat.copy()
+        for loc in batch_loc:
+            fn_args_flat[loc] = args_flat[loc][0]
+
+        fn_args = tree_unflatten(args_tree, fn_args_flat)
+
+        # Run 'fn' one time to get output-shape
+        init_result = fn(*fn_args, **kwargs)
+
+        # Return 'init_result' if none is requested
+        if not batch_size:
+            return init_result
+
+        init_result_flat, init_result_tree = tree_flatten(init_result)
+
+        out_axes_flat = _vmap_tree_flatten(out_axes)
+        out_pos = _get_batch_loc(out_axes_flat)
+        out_pos = out_pos[0]
+
+        num_axes_out = len(init_result_flat)
+
+        # Store batched results of all leaves
+        # in the flatten format with respect to the 'init_result' shape
+        batched_result_list = []
+        for j in range(num_axes_out):
+            out_shape = (
+                (batch_size,)
+                if not init_result_flat[j].shape
+                else (batch_size, *init_result_flat[j].shape)
+            )
+            batched_result_list.append(jnp.zeros(shape=out_shape, dtype=init_result_flat[j].dtype))
+            batched_result_list[j] = batched_result_list[j].at[0].set(init_result_flat[j])
+
+        if isinstance(out_axes, tuple) and len(out_axes) != len(out_shape):
+            raise ValueError(
+                "Invalid 'out_axes'; it can be an int or "
+                "a tuple of PyTrees corresponding to the "
+                f"arguments, but got {len(in_axes)} and {len(args)}"
+            )
```

Comment:
The error message here doesn't match the error message above which only mentions a single PyTree (rather than a tuple):
> "Invalid 'out_axes'; it can be a PyTree with integer leaves, but got {out_axes}"

The same goes for the type signature in the docstring.